platform/src/main/java/org/stellar/anchor/platform/job/FireblocksTransactionsReconciliationJob.java

```diff
@@ -40,44 +47,106 @@ public void reconcileTransactions() {
-        }
-        custodyTransactionService.updateCustodyTransaction(transaction);
+        handleStatusNotChanged(txn, attempt);
       }
     } catch (AnchorException | IOException e) {
-      errorEx(
-          String.format("Failed to reconcile status for transaction (id=%s)", transaction.getId()),
-          e);
+      errorEx(String.format("Failed to reconcile status for transaction (id=%s)", txn.getId()), e);
+    }
+  }
+
+  private void reconcileInboundTransactions(List<JdbcCustodyTransaction> transactions) {
+    if (transactions.isEmpty()) {
+      debug("No inbound transactions to reconcile");
+      return;
+    }
+
+    Instant startTime =
+        transactions.stream()
+            .map(JdbcCustodyTransaction::getCreatedAt)
+            .min(Instant::compareTo)
+            .orElse(null);
+
+    try {
+      List<TransactionDetails> fireblocksTransactions =
+          custodyPaymentService.getTransactionsByTimeRange(startTime, Instant.now());
+
+      if (fireblocksTransactions.isEmpty()) {
+        debug("No Fireblocks transactions within specified time range");
+        return;
+      }
+
+      Map<String, TransactionDetails> mappings =
+          fireblocksTransactions.stream()
+              .collect(
+                  Collectors.toMap(
+                      txn ->
+                          txn.getDestinationAddress() + StringUtils.SPACE + txn.getDestinationTag(),
+                      txn -> txn,
+                      (txn1, txn2) -> txn1));
+
+      transactions.forEach(
+          txn -> {
+            int attempt = txn.getReconciliationAttemptCount() + 1;
+
+            try {
+              String key = txn.getToAccount() + StringUtils.SPACE + txn.getMemo();
+              TransactionDetails fireblocksTxn = mappings.get(key);
+              if (fireblocksTxn != null && fireblocksTxn.getStatus().isObservable()) {
+                handleStatusChanged(fireblocksTxn, attempt);
+              } else {
+                handleStatusNotChanged(txn, attempt);
+              }
+            } catch (AnchorException | IOException e) {
+              errorEx(
+                  String.format("Failed to reconcile status for transaction (id=%s)", txn.getId()),
+                  e);
+            }
+          });
+    } catch (FireblocksException e) {
+      errorEx("Failed to retrieve fireblocks transactions", e);
+    }
+  }
+
+  private void handleStatusChanged(TransactionDetails fireblocksTxn, int attempt)
+      throws IOException, AnchorException {
+    debugF(
+        "Reconciliation attempt #[{}]: Fireblocks transaction status changed to [{}]",
+        attempt,
+        fireblocksTxn.getStatus().name());
+    Optional<CustodyPayment> payment = fireblocksEventService.convert(fireblocksTxn);
+    if (payment.isPresent()) {
+      fireblocksEventService.handlePayment(payment.get());
+    }
+  }
+
+  private void handleStatusNotChanged(JdbcCustodyTransaction txn, int attempt) {
```

Comment:
Should we update externalTxId here for inbound payments(just as we do for webhooks), but only if fireblocksTxn is not `null`?